internal/internal.go

```diff
@@ -230,19 +234,29 @@ func (p *envoyExtAuthzGrpcServer) eval(ctx context.Context, input ast.Value, opt
 
 		rs, err := rego.New(opts...).Eval(ctx)
 
-		// In "dry-run" mode, ignore all failure conditions
+		// In "dry-run" mode, we just log all failure conditions
 		// even ones that would typically be considered an error
-		if !p.cfg.DryRun {
-			if err != nil {
+		if err != nil {
```

Comment:
I think something like this might be simpler:

```go
if err != nil {
	err = fmt.Errorf("Error during Rego eval: %s", err.Error())
} else if len(rs) == 0 {
	err = fmt.Errorf("undefined decision")
} else if decision, ok = rs[0].Expressions[0].Value.(interface{}); !ok || len(rs) > 1 {
	err = fmt.Errorf("Decision was not interface{} OR len(rs) > 1")
}

if p.cfg.DryRun {
	logrus.Warnf(err.Error())
else {
	return err
} 
```

Reply:
ack, 1am coding was not a good idea